src/python/gem5/resources/resource.py

```diff
@@ -554,6 +555,70 @@ def _get_weights_and_simpoints_from_file(
         return simpoint_list, weight_list
 
 
+class SuiteResource(AbstractResource):
+    def __init__(
+        self,
+        workloads: List[Dict[str, Any]],
+        local_path: Optional[str] = None,
+        resource_version: Optional[str] = None,
+        description: Optional[str] = None,
+        source: Optional[str] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(
+            local_path=local_path,
+            description=description,
+            source=source,
+            resource_version=resource_version,
+        )
+        self.resource_version = resource_version
+        self.workloads = workloads
+        self.workload_obj_list = []
+        self.index = 0
+        self._local_path = local_path
+        self._description = description
+        self._source = source
+        self._obtain_workloads_list()
+
+    def _obtain_workloads_list(self) -> None:
+        for workload in self.workloads:
+            self.workload_obj_list.append(
+                WorkloadResource(
+                    workload["id"],
+                    resource_version=workload["resource_version"],
+                )
+            )
+
+    def __iter__(self) -> "SuiteResource":
```

Comment:
Doesn't this return a Workload?

Reply:
The return type should be a `Generator[YieldType, SendType, ReturnType]` (documentation: https://docs.python.org/3/library/typing.html#typing.Generator). We will change it to the correct type.